itenium-socks/src/app/socks/shop.component.ts

```diff
@@ -1,21 +1,78 @@
 import { Component } from '@angular/core';
 import { SocksService } from './socks.service';
-import { Observable } from 'rxjs';
+import { Observable,BehaviorSubject, combineLatest  } from 'rxjs';
 import { Sock } from './sock.model';
 import { AsyncPipe, NgFor } from '@angular/common';
+import { NgxPaginationModule } from 'ngx-pagination';
+import { map } from 'rxjs/operators';
+import { PricePipe } from '../price.pipe';
 
 @Component({
   selector: 'app-shop',
   standalone: true,
-  imports: [NgFor, AsyncPipe],
+  imports: [NgFor, AsyncPipe, NgxPaginationModule, PricePipe],
   templateUrl: './shop.component.html'
 })
 export class ShopComponent {
   socks$!: Observable<Sock[]>;
+  filteredSocks$!: Observable<Sock[]>;
+
+  public nameFilterSubject = new BehaviorSubject<string>('');
+  public colorFilterSubject = new BehaviorSubject<string>('');
+  public sortBySubject = new BehaviorSubject<string>('name');
+  public currentPageSubject = new BehaviorSubject<number>(1);
+  public pageSize = 10;
```

Comment:
Public is default, je hoeft dit dus niet speciaal te vermelden
Op sommige projecten doen ze dat wel --> volg dan de team/project standaarden